src/backend/distributed/operations/repair_shards.c

```diff
@@ -300,6 +300,14 @@ citus_move_shard_placement(PG_FUNCTION_ARGS)
 	Oid relationId = RelationIdForShard(shardId);
 	ErrorIfMoveCitusLocalTable(relationId);
 
+	bool waitForCleanupLock = false;
+
+	/*
+	 * We try to drop marked shards so that we won't unnecessarily error
+	 * for old shard placements.
+	 */
+	TryDropMarkedShards(waitForCleanupLock);
```

Comment:
should we wait for the cleanup lock here?

Reply:
done